### DISCUSSION TITLE
nextjs app dir experimental components missing in react-instantsearch-core

### DISCUSSION THREAD

**Discussion by duncanbla974** [algolia/instantsearch#5879] [Q&A]
I'm trying to use the experimental feature for the SSR with Nextjs 13 app dir but after installing packages
`yarn add react-instantsearch-nextjs` I'm getting error `./node_modules/react-instantsearch-nextjs/dist/es/InitializePromise.js
Attempted import error: 'useInstantSearchContext' is not exported from 'react-instantsearch-core` 
then I found out the experimental components in `node_modules/react-instantsearch-core/lib ` are missing even after reinstalling all instantsearch packages. Even if I'm installing react-instantsearch-core as I'm not in a react native projet.

Did i miss something ?

**Comment by dhayab** [comment]
This might be due to a mismatch between the versions of react-instantsearch* packages. Could you ensure all your packages have the same version number in your package.json file, and that your lockfile does not actually references differing version numbers?

**Reply by duncanbla974** [reply]
I have this in my package.json :
`"react-instantsearch": "7.1.0",
    "react-instantsearch-nextjs": "0.1.1",
    "react-instantsearch-router-nextjs": "7.1.0"` and in my yarn.lock :
    
   ```
 react-instantsearch-core@7.1.0:
  version "7.1.0"
  resolved "https://registry.yarnpkg.com/react-instantsearch-core/-/react-instantsearch-core-7.1.0.tgz#bf8ddc34655b5bd383a69f00979425ac5cc73c3f"
  integrity sha512-f+QZqUhQ1my/c//kw23Tn8jO1LQVqGRh8ObCbHnDzSPyI9Kp5+ETvI0BbXby1I4iObIjE1ApLhugBkli77zsTQ==
  dependencies:
    "@babel/runtime" "^7.1.2"
    algoliasearch-helper "3.14.2"
    instantsearch.js "4.57.0"
    use-sync-external-store "^1.0.0"
```

```
react-instantsearch-nextjs@^0.1.1:
  version "0.1.1"
  resolved "https://registry.yarnpkg.com/react-instantsearch-nextjs/-/react-instantsearch-nextjs-0.1.1.tgz#07f6192e417e7cb53d33f8f8cbff94214baa0c2e"
  integrity sha512-2Y/GH8vVowTnvGVk4C8njwxPkDgIrOAReMf/vkI05DLTmrKUEVHsn3fvSB9p8ozv0C+teNxd3/Vv4ghbwttNpw==

```
```
react-instantsearch-router-nextjs@7.1.0:
  version "7.1.0"
  resolved "https://registry.yarnpkg.com/react-instantsearch-router-nextjs/-/react-instantsearch-router-nextjs-7.1.0.tgz#327381595217a804da81363894798f62194131f6"
  integrity sha512-4ot0ZM/jbWXdnBatxBedRVRuJGruDpC248ukOUJHbdexiBPTIZYZPHz4icmFXxJvqlwRlTPjiihx5evnlMqLYA==
  dependencies:
    instantsearch.js "4.57.0"
    react-instantsearch-core "7.1.0"
```

```
react-instantsearch@7.1.0:
  version "7.1.0"
  resolved "https://registry.yarnpkg.com/react-instantsearch/-/react-instantsearch-7.1.0.tgz#812bd7c28ccdd08cc4e6087e07c144eb85097dee"
  integrity sha512-AdOP2+ubNVZCkODi45IXpWo9AklTjEGm9xqFQ227zt6CA6NLVhY5otFGkWPBrQwwbEnSKO5PH563pfNGnCMsRQ==
  dependencies:
    "@babel/runtime" "^7.1.2"
    instantsearch.js "4.57.0"
    react-instantsearch-core "7.1.0"

```

**Reply by Haroenv** [reply]
Those versions look correct. I see you closed the discussion @duncanbla974, did you find out what's going on?

**Reply by duncanbla974** [reply]
@Haroenv Yes I did rebuild my docker container and it works 👍🏼
